zipline/data/data_portal.py

```diff
@@ -429,24 +429,25 @@ def _get_minute_spot_value_future(self, asset, column, dt):
         else:
             return result
 
-    def _get_minute_spot_value(self, asset, column, dt):
+    def _get_minute_spot_value(self, asset, column, dt, ffill=False):
         result = self._equity_minute_reader.get_value(
-            asset.sid,
-            dt,
-            column
+            asset.sid, dt, column
         )
 
-        if result > 0 or column == "volume":
-            return result
+        if result == 0 or np.isnan(result):
+            if column == "volume":
```

Comment:
I'd recommend restructuring so that volume and OHLC handling are not comingled.
With the implementation above when `volume` is non-zero `np.isnan(result)` is also called.

That extra call would be removed with:

```
if column == "volume":
     if result == 0:
        return 0
elif np.isnan(result) and not ffill:
       return  np.nan
```